### PR TITLE
docs: add docs for banking example

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Primary Reviewer

@autodidaddict 

## Feature or Problem

This PR adds documentation for the banking example

## Related Issues

N/A

## Release Information

`next`

## Consumer Impact

Consumers will be able to more easily try out concordance

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [x] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [x] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Unit Test(s)

N/A

### Acceptance or Integration

N/A

### Manual Verification

Verified locally

### CODE REVIEW

**Review by connorsmith256 — CHANGES_REQUESTED** [review]
This is great. Just a couple small things I ran into:
- the `./docs/videos/all-links-established.png` link is broken for me
- I don't think the event sourcing link defs are quite right. I'm getting these errors from the provider:

```
2023-04-11T15:59:21.637452Z ERROR put_link{actor_id=MC5EQZ6NZY2T5US5JJTCJVAWHETCIIZLLVFUPERSTO2T3AR2NF62JWKI provider_id=VC2SOX5BVQFQ7MR763QWGVTPVMUZPMB7WW23MQTQIUMQWS6OCLHJADFW link_name=default}: concordance_es_provider::wcprovider: Failed to derive interest declarations from link definition. Aborting due to error: Failed to parse valid interest declaration from link definition: LinkDefinition { actor_id: "MC5EQZ6NZY2T5US5JJTCJVAWHETCIIZLLVFUPERSTO2T3AR2NF62JWKI", provider_id: "VC2SOX5BVQFQ7MR763QWGVTPVMUZPMB7WW23MQTQIUMQWS6OCLHJADFW", link_name: "default", contract_id: "wasmcloud:eventsourcing", values: {"\"stop\":[\"interbank_transfer_completed\"": "\"interbank_transfer_failed\"]}'", "\"advance\":[\"wire_funds_reserved\"": "\"interbank_transfer_initiated\"]", "ROLE": "process_...
```

```
2023-04-11T15:59:36.908062Z ERROR put_link{actor_id=MCZ2V2VTF4S4QAYKHJTGARIGWFMQXS2FDHKNNI3H7ZHHYAWE6IVCTD7M provider_id=VC2SOX5BVQFQ7MR763QWGVTPVMUZPMB7WW23MQTQIUMQWS6OCLHJADFW link_name=default}: concordance_es_provider::wcprovider: Failed to derive interest declarations from link definition. Aborting due to error: Failed to parse valid interest declaration from link definition: LinkDefinition { actor_id: "MCZ2V2VTF4S4QAYKHJTGARIGWFMQXS2FDHKNNI3H7ZHHYAWE6IVCTD7M", provider_id: "VC2SOX5BVQFQ7MR763QWGVTPVMUZPMB7WW23MQTQIUMQWS6OCLHJADFW", link_name: "default", contract_id: "wasmcloud:eventsourcing", values: {"ROLE": "aggregate KEY=account_number INTEREST=bankaccount NAME=bankaccount"} }
```

I think the second one is missing a close quote on the value for `ROLE`. I'm not sure why the first one is failing. I copied these values from the table in the walkthrough into the washboard

When I copy-pasted the script that uses wash to set the link defs, it worked 🤔 

Ah, now I remember: the washboard tries to parse link def values by splitting on commas. The process manager and aggregate values are using spaces, so those should be commas. However, the JSON for the process manager's `INTEREST` field has commas in it, so I'm not sure the washboard will be able to process that at all. It might be better to drop the table and just have the script, or at least call out that the process manager needs to be set up with wash 😔 

- I can't get commands to turn into events. When I submit a command I see it in the command stream, but nothing in the events stream. I don't see any errors from the provider either

**Comment by connorsmith256** [comment]
Maybe things _are_ working(?) I expected to see messages in the `CC_EVENTS` stream, but it remains empty. However, the aggregate state is updating correctly

**Comment by connorsmith256** [comment]
I also double-checked that redis is updating. Maybe add a `redis-cli get balance.ABC123` to show that the projection state is getting updated too?

File: examples/bankaccount/README.md [hunk 1]
```diff
@@ -1,4 +1,294 @@
...
+```console
+sudo apt install wash
+```
+
+For other options on how to install `wash`, see [the `wasmcloud/wash`][wash-docs-install].
+
+Along with `wash`, ensure you have access to the following:
+
+- [`nats` client](https://docs.nats.io/running-a-nats-service/clients) for communicating with NATS
+- [`jq`](https://stedolan.github.io/jq/) for displaying JSON
```
Comment (connorsmith256) [inline examples/bankaccount/README.md:36]: I think we need to mention having redis installed/running

File: examples/bankaccount/README.md [hunk 2]
```diff
@@ -1,4 +1,294 @@
...
+| `projector`       | `concordance` | `default` | `wasmcloud:eventsourcing` | `ROLE=projector,INTEREST=account_created,funds_deposited,funds_withdrawn,wire_funds_reserved,wire_funds_released,NAME=bankaccount_projector`                                                                                                       |
+| `projector`       | `keyvalue`    | `default` | `wasmcloud:keyvalue`      | `URL='redis://0.0.0.0:6379/'`                                                                                                                                                                                                                      |
+| `process_manager` | `concordance` | `default` | `wasmcloud:eventsourcing` | `ROLE=process_manager KEY=wire_transfer_id NAME=interbankxfer INTEREST='{"start":"wire_transfer_requested","advance":["wire_funds_reserved","interbank_transfer_initiated"],"stop":["interbank_transfer_completed","interbank_transfer_failed"]}'` |
```
Comment (connorsmith256) [inline examples/bankaccount/README.md:159]: These should be comma-separated for consistency, or the projector link should use spaces, or perhaps both should use newlines

File: examples/bankaccount/README.md [hunk 3]
```diff
@@ -1,4 +1,294 @@
...
+    cosmonic:eventsourcing \
+    ROLE=process_manager KEY=wire_transfer_id NAME=interbankxfer INTEREST='{"start":"wire_transfer_requested","advance":["wire_funds_reserved","interbank_transfer_initiated"],"stop":["interbank_transfer_completed","interbank_transfer_failed"]}'
+
+# Link aggregate <-> concodrance
+wash ctl link put $AGGREGATE_ACTOR_ID $CONCORDANCE_PROVIDER_ID \
+    cosmonic:eventsourcing \
+    ROLE=aggregate KEY=account_number INTEREST=bankaccount NAME=bankaccount
+```
+
+> **Warning**
```
Comment (connorsmith256) [inline examples/bankaccount/README.md:191]: This is easy to miss. How about moving the script and warning above the table, and reword the preface for the table to something like "after running the script, here is the config we set:"

**Review by connorsmith256 — CHANGES_REQUESTED** [review]
Just a couple more nits and we're good to go

File: examples/bankaccount/README.md (hunk 1 above)
Comment (vados-cosmonic) [reply examples/bankaccount/README.md:36]: - [ ] mention `redis-server`

File: examples/bankaccount/README.md (hunk 3 above)
Comment (vados-cosmonic) [reply examples/bankaccount/README.md:191]: - [ ] Move warning and script before table

**Review by connorsmith256 — APPROVED** [review]
:shipit:

**Pull request merged**